src/containerapp/azext_containerapp/custom.py

```diff
@@ -1870,3 +1873,28 @@ def init_dapr_components(cmd, resource_group_name, environment_name, statestore=
             "daprComponents": [statestore_component_id, pubsub_component_id]
         }
     }
+
+def set_environment_telemetry(cmd,
+                                 name,
+                                 resource_group_name,
+                                 app_insights_connection_string = None,
+                                 open_telemetry_traces_destinations = None,
+                                 open_telemetry_logs_destinations = None,
+                                 open_telemetry_metrics_destinations = None,
+                                 open_telemetry_dataDog_site = None,
+                                 open_telemetry_dataDog_key = None):
+    raw_parameters = locals()
+    containerapp_env_telemetry_decorator = ContainerappEnvTelemetryPreviewSetDecorator(
+        cmd=cmd,
+        client=ManagedEnvironmentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    containerapp_env_telemetry_decorator.validate_arguments()
+    containerapp_env_telemetry_decorator.register_provider(CONTAINER_APPS_RP)
+
+    containerapp_env_telemetry_decorator.construct_payload()
+    r = containerapp_env_telemetry_decorator.update()
+    r = containerapp_env_telemetry_decorator.post_process(r)
```

Comment:
You can remove unnecessary execution steps, for this command, the `post_process` is not needed.